kgforge/core/conversions/dataframe.py

```diff
@@ -65,24 +65,51 @@ def _from_dataframe(row: Series, na: Union[Any, List[Any]], nesting: str) -> Res
 
 
 def deflatten(items: List[Tuple[str, Any]], sep: str) -> Dict:
-    d = {}
-    i = 0
-    while i < len(items):
-        k, v = items[i]
-        if sep not in k:
-            d[k] = v
-            i += 1
+    """
+
+
+    Parameters
+    ----------
+    items : List[Tuple[str, Any]]
+        at the beginning (first call in the recursion cycle) it is obtain within _from_dataframe as a list, one per row,
+        of tuples where the first item is the column name and the second is the value
+    sep : str
+        the separator. Usually '.'
+
+    Raises
+    ------
+    ValueError
+        One cannot provide both 'property.subproperty1' and 'property'. In such case, a ValueError is raised
+
+    Returns
+    -------
+    Dict
+        a deflattened (nested) dictionary of {columnName: value}.
+        Deflatten means that the separator implies nesting of the dictionary.
+
+    """
+    d = {}  # dictionary that will be returned
+    split = []  # those we have already split (we need to keep track)
+    for n1, t1_ in enumerate(items):
+        if n1 in split:  # already done, continue
+            continue
+        k1, v1 = t1_  # all tuples refer to a key-value pair in the DataFrame row
+        if isinstance(k1, str) and sep in k1:  # avoid issue if int or other class
```

Comment:
I think this check that it is a string is also applicable to the else statement. So there should be an if that makes sure k1 is a str, otherwise an error is thrown. And then within that if you have your current if-else  based on the presence of the separator. 

Unless you would want something other than a string in your else statement?

Reply:
Actually the is instance allows for non-strings to be processed in the else statement.
I do not think any resource should have non-string attributes, but I think that the error should not be raised by this function, but (if at all) by the Resource.from_json.
I see the following options:

- no check that it is a string => a non-string (e.g. int) would raise a generic error when we check if sep is in it. This is the behaviour of the master branch as of now
- check that this is a string in the if, not in the else => a non-string key in items is processed and returned in the dictionary. When the latter is passed to forge.from_json, it might raise an error
- check in both the if and the else => the non-string key and its value are discarded without any message. I do not think that this is desirable.
- we could raise a specific ValueError if any key in the key-value pairs is not a string

Reply:
I would have an initial check that all keys are strings initially, without even getting into the details. 

`if not all(isinstance(k, str) for k, _ in items): raise something`